blocks/theano_expressions.py

```diff
@@ -14,11 +14,11 @@ def l2_norm(tensors):
         The tensors.
 
     """
-    flattened = [tensor.as_tensor_variable(t).flatten() for t in tensors]
-    flattened = [(t if t.ndim > 0 else t.dimshuffle('x'))
-                 for t in flattened]
-    joined = tensor.join(0, *flattened)
-    return tensor.sqrt(tensor.sqr(joined).sum())
+    grand_sum = 0.0
```

Comment:
I wrote to the mailing list. Maybe they will pick it up.

Dmitry Bogdanov wrote:
>
> In blocks/theano_expressions.py
> <https://github.com/mila-udem/blocks/pull/781#discussion_r36043448>:
>
> > @@ -14,11 +14,11 @@ def l2_norm(tensors):
> > The tensors.
> >
> > """
> > - flattened = [tensor.as_tensor_variable(t).flatten() for t in tensors]
> > - flattened = [(t if t.ndim> 0 else t.dimshuffle('x'))
> > - for t in flattened]
> > - joined = tensor.join(0, *flattened)
> > - return tensor.sqrt(tensor.sqr(joined).sum())
> > + grand_sum = 0.0
>
> Wow! You should probably get in touch with Theano devs if you have
> such an amazing fix for them.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mila-udem/blocks/pull/781/files#r36043448>.